plugins/modules/ipaautomountkey.py

```diff
@@ -0,0 +1,217 @@
+        key = self.get_key(self.ipa_params.location,
+                           self.ipa_params.mapname,
+                           self.ipa_params.key)
+
+        if self.ipa_params.state == "present":
+            if key is None:
+                # does not exist and is wanted
+                args["automountinformation"] = self.ipa_params.info
+                self.add_ipa_command(
+                    "automountkey_add",
+                    name=self.ipa_params.location,
+                    args=args
+                )
+            elif key is not None:
+                # exists and is wanted, check for changes
+                if self.ipa_params.info != \
+                        key.get('automountinformation', [None])[0]:
+                    args["newautomountinformation"] = self.ipa_params.info
+                    self.add_ipa_command(
+                        "automountkey_mod",
+                        name=self.ipa_params.location,
+                        args=args
+                    )
+        elif self.ipa_params.state == "rename":
+            if key is not None:
+                newkey = self.get_key(self.ipa_params.location,
+                                      self.ipa_params.mapname,
+                                      self.ipa_params.newname)
+
+                args["rename"] = self.ipa_params.newname
+                if newkey is None:
+                    self.add_ipa_command(
+                        "automountkey_mod",
+                        name=self.ipa_params.location,
+                        args=args
+                    )
+        else:
+            # if key exists and self.ipa_params.state == "absent":
+            if key is not None:
+                self.add_ipa_command(
+                    "automountkey_del",
+                    name=self.ipa_params.location,
+                    args=args
+                )
+
+
+def main():
+    ipa_module = AutomountKey(
+        argument_spec=dict(
+            ipaadmin_principal=dict(type="str",
+                                    default="admin"
+                                    ),
+            ipaadmin_password=dict(type="str",
+                                   required=False,
+                                   no_log=True
+                                   ),
+            state=dict(type='str',
+                       default='present',
+                       choices=['present', 'absent', 'rename']
+                       ),
+            location=dict(type="str",
+                          aliases=["automountlocationcn", "automountlocation"],
+                          default=None,
+                          required=True
+                          ),
+            newname=dict(type="str",
+                         aliases=["newkey", "new_name",
+                                  "new_key", "newautomountkey"],
+                         default=None,
+                         required=False
+                         ),
```

Comment:
This is where I think the ansible-freeipa module should diverge a little from the CLI commands.

All ansible-freeipa modules which support state: renamed support variable rename, with new_name as an alias, and these should be the default for renaming the object.

I'm fine with newkey and newautomontkey being kept as aliases, but not newname.